app/code/Magento/Sales/Model/InvoiceOrder.php

```diff
@@ -158,11 +159,11 @@ public function execute(
             $arguments
         );
         if ($errorMessages->hasMessages()) {
+            $connection->rollBack();
```

Comment:
I'd recommend to refactor the implementation to guarantee the symmetric transaction start/rollback. Moving the validation under the `try..catch` block below would eliminate the need to duplicate `$connection->rollBack()` calls.